scripts/build_qat_envoy.sh

```diff
@@ -0,0 +1,201 @@
+#!/usr/bin/env bash
+#
+# Copyright (c) 2019 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+set -e
+
+if [ -f /etc/os-release ]; then
+	source /etc/os-release
+else
+	source /usr/lib/os-release
+fi
+
+readonly BAZEL_BIN="$(realpath "${PWD}"/bazel.sh)"
+readonly SCRIPTS_DIR="$(realpath "$(dirname "$0")")"
+readonly VERSIONS_FILE="${SCRIPTS_DIR}/../versions.yaml"
+readonly OPENSSL_DIR="${SCRIPTS_DIR}/../openssl"
+readonly QAT_ENGINE_DIR="${SCRIPTS_DIR}/../QAT_Engine"
+readonly ENVOY_DIR="${SCRIPTS_DIR}/../envoy"
+readonly QAT_LIB_DIR="${SCRIPTS_DIR}/../QAT_Lib"
+
+#shellcheck source=lib.sh
+source "${SCRIPTS_DIR}/lib.sh"
+
+versions_yaml() {
+	shyaml get-value "$1" < "${VERSIONS_FILE}"
+}
+
+install_bazel() {
+	info "Download and install bazel"
+	local bazel_url
+	local bazel_version
+	bazel_url="$(versions_yaml "tools.bazel.url")"
+	bazel_version="$(versions_yaml "tools.bazel.version")"
+	bazel_sha256="$(versions_yaml "tools.bazel.sha256")"
+	curl -L "${bazel_url}/releases/download/${bazel_version}/bazel-${bazel_version}-installer-linux-x86_64.sh" -o "${BAZEL_BIN}"
+	bazel_sha256_sum="$(sha256sum "${BAZEL_BIN}" | cut -d' ' -f1)"
+	# check sha256
+	if [ "${bazel_sha256_sum}" != "${bazel_sha256}" ]; then
+		die "Mismatch bazel sha256. Expecting ${bazel_sha256}, got ${bazel_sha256_sum}"
+	fi
+
+	chmod +x "${BAZEL_BIN}"
+}
+
+setup() {
+	case $ID in
+		clear-linux*)
+			export HOME=/root
+			info "Clear Linux OS detected"
+			rm -rf /run/lock/clrtrust.lock
+			clrtrust generate
+			swupd update
+			swupd bundle-add os-core-dev
+			;;
+		debian|ubuntu)
+			info "Debian/Ubuntu OS detected"
+			# https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=863199
+			mkdir -p /usr/share/man/man1
+			apt-get -o Acquire::Check-Valid-Until=false update
+			apt-get -y install git libtool \
+					cmake clang-format-7 automake ninja-build curl \
+					git build-essential wget libudev-dev libssl-dev \
+					openssl pkg-config autoconf autogen libtool \
+					libssl-dev pkg-config zip g++ zlib1g-dev unzip \
+					python python-pip curl gnupg2
+			;;
+	esac
+
+	pip install shyaml
+
+	install_bazel
+	"${BAZEL_BIN}" --user
+}
+
+build_install_qat_library() {
+	local qat_url
+	local qat_version
+
+	qat_url="$(versions_yaml "libraries.qat.url")"
+	qat_version="$(versions_yaml "libraries.qat.version")"
+
+	mkdir -p "${QAT_LIB_DIR}"
+	pushd "${QAT_LIB_DIR}"
+
+	curl -L "${qat_url}/qat${qat_version}.tar.gz" -o qat.tz
```

Comment:
I think it should be `qat.tgz` even though it doesn't matter that much. You should do the SHA256 check here too.